python/pyspark/sql/functions/builtin.py

```diff
@@ -20681,6 +20681,26 @@ def nullif(col1: "ColumnOrName", col2: "ColumnOrName") -> Column:
     return _invoke_function_over_columns("nullif", col1, col2)
 
 
+@_try_remote_functions
+def nullifzero(col: "ColumnOrName") -> Column:
+    """
+    Returns null if `col` is equal to zero, or `col` otherwise.
+
+    .. versionadded:: 4.0.0
+
+    Parameters
+    ----------
+    col : :class:`~pyspark.sql.Column` or str
+
+    Examples
+    --------
+    >>> df = spark.createDataFrame([(0,), (1,)], ["a"])
+    >>> df.select(nullifzero(df.a).alias('r')).collect()
+    [Row(r=None), Row(r=1)]
```

Comment:
Could we add a test in the actual `test_functions.py`? Then the test will be automatically shared between Spark Connect and Spark Classic